### PR TITLE
feat(inbox): Move saved search dropdown to tab (WOR-540)

### PR DESCRIPTION
**Pull request by scttcper**
Moves the saved search dropdown to a tab with a dropdown. Will highlight the tab as "custom search" when the search doesn't match any of the tabs.

Changes tabs to links that can be middle clicked to open in a new browser tab.

Custom search tab highlighted:
image

Saved search dropdown:
image

### CODE REVIEW

File: src/sentry/static/sentry/app/views/issueList/savedSearchMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
+import React from 'react';
```
Comment (scttcper) [inline src/sentry/static/sentry/app/views/issueList/savedSearchMenu.tsx:1]: The savedSearchMenu is now just the dropdown components shared between the original dropdown button and the savedSearch tab being added in this PR

File: src/sentry/static/sentry/app/views/issueList/savedSearchTab.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        />
+      </StyledDropdownLink>
+    </TabWrapper>
+  );
+}
+
+export default SavedSearchTab;
+
+const TabWrapper = styled('li')<{isActive?: boolean}>`
+  /* Color matches nav-tabs - overriten using dark mode class saved-search-tab */
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/issueList/savedSearchTab.tsx:56]: nit: overwritten*

File: src/sentry/static/sentry/app/views/issueList/header.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import {InjectedRouter} from 'react-router';
+import {InjectedRouter, Link as RouterLink} from 'react-router';
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/issueList/header.tsx:2]: is the rename just to make the naming more explicit?

**Review by davidenwang — APPROVED** [review]
lgtm

**Pull request merged**